src/esportsbot/cogs/MusicCog.py

```diff
@@ -0,0 +1,1117 @@
+                self.__play_queue(guild_id)
+
+            await self.__update_channel_messages(guild_id)
+            return True
+        except Exception as e:
+            print("There was an error while adding to the queue: \n" + str(e))
+            return False
+
+    async def on_message_handle(self, message: Message):
+        """
+        The handle the is called whenever a message is sent in the music channel of a guild.
+        :param message: The message sent to the channel.
+        :return: None
+        """
+
+        if message.content.startswith(self._bot.command_prefix):
+            # Ignore commands, any MusicCog commands will get handled in the usual way
+            return
+
+        if not message.author.voice:
+            # User is not in a voice channel.. exit
+            send = Embed(title="You must be in a voice channel to add a song", colour=EmbedColours.orange())
+            await message.delete()
+            await self.__send_timed_message(message.channel, send, timer=10)
+            return
+
+        if not message.author.voice.channel.permissions_for(message.guild.me).connect:
+            # The bot does not have permission to join the channel.. exit
+            send = Embed(title="I need the permission `connect` to be able to join that channel",
+                         colour=EmbedColours.orange())
+            await message.delete()
+            await self.__send_timed_message(message.channel, send, timer=10)
+            return
+
+        if not self._currently_active.get(message.guild.id):
+            # We aren't in a voice channel in the given guild
+            voice_client = await message.author.voice.channel.connect()
+            self.__add_new_active_channel(message.guild.id, voice_client=voice_client,
+                                          channel_id=message.author.voice.channel.id)
+        else:
+            if self._currently_active.get(message.guild.id).get('channel_id') != message.author.voice.channel.id:
+                # The bot is already being used in the current guild.
+                send = Embed(title="I am already in another voice channel in this server", colour=EmbedColours.orange())
+                await message.delete()
+                await self.__send_timed_message(message.channel, send, timer=10)
+                return
+
+        # Check if the loops for marked and active channels are running.
+        self.__check_loops_alive()
+
+        # Splits multiline messages into a list. Single line messages return a list of [message]
+        split_message = message.content.split("\n")
+        split_message = list(filter(lambda k: k.replace(' ', '') != '', split_message))
+        partial_success = True
+
+        # Add each line of the message to the queue
+        st = time.time()
+        for line in split_message:
+            partial_success = await self.process_song_request(message, line) or partial_success
+        print('Queue time:' + str(time.time() - st))
```

Comment:
for usability, you may wish to use `datetime` and `timedelta`. E.g:
```
st = datetime.now()
...
print('Queue time:' + lib.timeUtil.td_format_noYM(datetime.now() - st))
```
`td_format_noYM` gives you strings like `2 days, 10 hours, 5 minutes". Added in https://github.com/FragSoc/esports-bot/pull/29

Obviously _less_ important for console debugging, but great for user-facing things.

Reply:
I can implement this once the merge has been made as the my current branch doesn't have lib.timeUtils